src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/SingleShot.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            CreateSingleShot(modelAbsPath, inTensorsInfo, outTensorsInfo);
+        }
+
+        /// <summary>
+        /// Destructor of the Single instance.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
+        ~SingleShot()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Invokes the model with the given input data.
+        /// </summary>
+        /// <param name="inTensorsData">The input data to be inferred.</param>
+        /// <returns>TensorsData instance which contains the inferred result.</returns>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="IOException">Thrown when failed to push an input data into source element.</exception>
+        /// <exception cref="TimeoutException">Thrown when failed to get the result from sink element.</exception>
+        /// <since_tizen> 5.5 </since_tizen>
+        public TensorsData Invoke(TensorsData inTensorsData)
+        {
+            TensorsData out_data;
+            IntPtr out_ptr;
+            NNStreamerError ret = NNStreamerError.None;
+
+            ret = Interop.SingleShot.InvokeSingle(_handle, inTensorsData.Handle, out out_ptr);
+            NNStreamer.CheckException(ret, "fail to invoke the single inference engine");
+
+            out_data = TensorsData.CreateFromNativeHandle(out_ptr);
+            return out_data;
+        }
+
+        private void CreateSingleShot(string modelAbsPath, TensorsInfo inTensorInfo, TensorsInfo outTensorInfo)
+        {
+            NNStreamerError ret = NNStreamerError.None;
+            IntPtr input_info;
+            IntPtr output_info;
+
+            /* Check model path */
+            if (string.IsNullOrEmpty(modelAbsPath))
+                ret = NNStreamerError.InvalidParameter;
+            NNStreamer.CheckException(ret, "model path is invalid: " + modelAbsPath);
+
+            input_info = inTensorInfo.GetTensorsInfoHandle();
+            output_info = outTensorInfo.GetTensorsInfoHandle();
+
+            ret = Interop.SingleShot.OpenSingle(out _handle, modelAbsPath, input_info, output_info, NNFWType.Any, HWType.Any);
+            NNStreamer.CheckException(ret, "fail to open the single inference engine");
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object. Can also dispose any other disposable objects.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        protected virtual void Dispose(bool disposing)
```

Comment:
https://docs.microsoft.com/ko-kr/dotnet/standard/garbage-collection/implementing-dispose

_handle is umanaged object